R/drive_cp.R

```diff
@@ -0,0 +1,118 @@
+#' Copy a Google Drive file.
+#'
+#' `drive_copy()` is an alias for `drive_cp()`.
+#'
+#' @seealso Wraps the
+#' [drive.files.copy](https://developers.google.com/drive/v3/reference/files/copy)
+#' endpoint
+#'
+#' @template file
+#' @template path
+#' @param name Character, new file name if not specified as part of `path`. Any
+#'   name obtained from `path` overrides this argument. Defaults to "Copy of
+#'   {CURRENT-FILE-NAME}."
+#' @template verbose
+#' @template dribble-return
+#'
+#' @examples
+#' \dontrun{
+#' ## create a file to copy
+#' file <- drive_upload(system.file("DESCRIPTION"), "DESC")
+#'
+#' ## Make a "Copy of" copy in same folder as the original
+#' drive_cp("DESC")
+#'
+#' ## Make an explicitly named copy in same folder as the original
+#' drive_cp("DESC", "DESC-two")
+#'
+#' ## Make an explicitly named copy in a different folder
+#' folder <- drive_mkdir("new-folder")
+#' drive_cp("DESC", folder, "DESC-three")
+#'
+#' ## Behold all of our copies!
+#' drive_search("DESC")
+#'
+#' ## Delete all of our copies and the new folder!
+#' drive_search("DESC") %>% drive_delete()
+#' drive_delete(folder)
+#' }
+#' @export
+drive_cp <- function(file = NULL, path = NULL, name = NULL,  verbose = TRUE) {
+  file <- as_dribble(file)
+  file <- confirm_single_file(file)
+  if (is_folder(file)) {
+    stop("The Drive API does not copy folders.", call. = FALSE)
+  }
+
+  if (!is.null(name)) {
+    stopifnot(is_path(name), length(name) == 1)
+  }
+
+  if (is_path(path)) {
+    if (is.null(name)) {
+      path_parts <- partition_path(path)
```

Comment:
Once I remove the last use of `split_path_name()`, I'm going to delete it. I think it's better for a helper like that to be less flexible: take only one input (a path vs path *and* name) and only one type of input (path as character vs all forms of path).